core/src/wallet/tezos/delegation/TezosLikeOriginatedAccount.cpp

```diff
@@ -142,19 +145,24 @@ namespace ledger {
         }
 
         FuturePtr<api::Amount> TezosLikeOriginatedAccount::getBalance(const std::shared_ptr<api::ExecutionContext>& context) {
-            return std::dynamic_pointer_cast<OperationQuery>(queryOperations()->complete())->execute()
-                .mapPtr<api::Amount>(context, [] (const std::vector<std::shared_ptr<api::Operation>> &ops) {
-                    auto result = BigInt::ZERO;
-
-                    for (auto const &op : ops) {
-                        OperationStrategy::update_balance(op, result);
-                    }
-
-                    return std::make_shared<Amount>(currencies::TEZOS, 0, result);
+            auto localAccount = _originatorAccount.lock();
+            if (!localAccount) {
+                throw make_exception(api::ErrorCode::NULL_POINTER, "Account was released.");
+            }
+            auto wallet = std::dynamic_pointer_cast<TezosLikeWallet>(localAccount->getWallet());
```

Comment:
we should check for null pointer here to avoid segfaults later

Reply:
The only case that I can see to get a sigfault here is to have a TezosLikeAccount inside a wallet different from a TezosLikeWallet, which is not possible.
I know that we should check for null after a dynamic cast when we are not sure about type matching. But here we are sure about the pointer type.